pkg/apiserver/apiserver.go

```diff
@@ -165,3 +169,26 @@ func (c completedConfig) New() (*APIServer, error) {
 
 	return s, nil
 }
+
+// CleanupDeprecatedAPIServices deletes the registered APIServicve resources for
+// the deprecated Antrea API groups.
+func CleanupDeprecatedAPIServices(aggregatorClient clientset.Interface) error {
+	// The APIService of a deprecated API group should be added to the slice.
+	// After Antrea upgrades from an old version to a new version that
+	// deprecates a registered APIService, the APIService should be deleted,
+	// otherwise K8s will reject to delete an existing Namespace created
+	// before the upgrade. Also check the information at:
```

Comment:
suggestion: s/will reject to delete/will fail to delete

also question: is it just for Namespaces that existed before the upgrade? I thought that was for all Namespaces (because every time a Namespace is deleted, K8s reaches out to all APIServices to see if they have any object that needs to be deleted). See https://github.com/kubernetes/kubernetes/issues/60807#issuecomment-455069171. I may be wrong though.

Reply:
Changed to "will fail to delete".

You are right. It should be all Namespaces before the APIService is deleted. Updated the comments.